### PR TITLE
Adding Concat, Split and Upsample layers

### PR DESCRIPTION
**Pull request by stefkohub**
I have added `to_axon_concat`, `to_axon_split` and `to_axon_upsample` functions. 
I think there is a quite more elegant solution to write the `to_axon_upsample` function but I cannot find it at the moment.
Unlike the other functions, the `to_axon_split` function is returning more than one layer so I used Enum.reduce looping on new layers which are updating the axon set of layers.

### CODE REVIEW

File: lib/axon_onnx/deserialize.ex [hunk 1]
```diff
@@ -658,6 +667,69 @@ defmodule AxonOnnx.Deserialize do
...
+         %Node{attribute: attrs, input: [inp, scale], output: [output_name]},
+         axon,
+         params,
+         used_params
+       ) do
+    inp = input_or_param!(inp, params, axon, used_params)
+    scale = input_or_param!(scale, params, axon, used_params)
+    updated_params = 
+      used_params
+      |> Map.put(output_name <> "_scale", scale)
```
Comment (seanmor5) [inline lib/axon_onnx/deserialize.ex:698]: So the problem here is that `scales` will actually be updated during the training process - which is something we don't want. Looking at `Resize`, they mark `scales` as non-differentiable, which means they are constant. So we'll want to make these a constant part of the model as well. You can just remove this altogether then - just a note that I will need to update a few other layers to follow this convention as well.

```suggestion
```

File: lib/axon_onnx/deserialize.ex [hunk 2]
```diff
@@ -658,6 +667,69 @@ defmodule AxonOnnx.Deserialize do
...
+    unless [1.0, 1.0 | scale] = scale|>Nx.to_flat_list do
+      raise "Wrong scale format. First two values must be [1, 1]."
+    end
+    [_, _ | inp_output_shape] = Tuple.to_list(inp.output_shape)
```
Comment (seanmor5) [inline lib/axon_onnx/deserialize.ex:702]: First, you can match directly on the input to extract shape with:

```elixir
%Axon{output_shape: shape} = inp = input_or_param!(...)
```

Then we can refactor:

```suggestion
    output_shape =
      case Nx.to_flat_list(scale) do
        [1.0, 1.0 | scale] ->
          output_shape =
            scale
            |> Enum.zip_with(shape, fn x, y -> floor(x * y) end)
            |> List.to_tuple()
   
        [s1, s2 | scale] ->
          raise ArgumentError, "unspported scale format, first two scale values must be 1, got #{s1} and #{s2}"
```

File: lib/axon_onnx/deserialize.ex [hunk 3]
```diff
@@ -658,6 +667,69 @@ defmodule AxonOnnx.Deserialize do
...
+    %{ "mode" => mode } = options!(attrs)
+
+    output_shape = List.to_tuple(
+      for {x, y} <- List.zip([scale, inp_output_shape]), do: floor x*y
+    )
```
Comment (seanmor5) [inline lib/axon_onnx/deserialize.ex:707]:
```suggestion
```

**Review by seanmor5 — COMMENTED** [review]
Nice! Just a few minor changes

File: lib/axon_onnx/deserialize.ex (hunk 2 above)
Comment (stefkohub) [reply lib/axon_onnx/deserialize.ex:702]: Oh I see, thank you for this, it is far more elegant than mine :)

**Comment by stefkohub** [comment]
Ok I'm going to apply the changes with your suggestions, thank you!

**Comment by seanmor5** [comment]
@stefkohub Can you please rebase off the master branch? It should fix the failing CI/CD

**Comment by stefkohub** [comment]
Hi @seanmor5 I have merged my branch with your master and now the CI/CD is back to working.

(P.S. Happy new year!)

**Pull request merged**